src/Link/index.tsx

```diff
@@ -145,22 +153,54 @@ export default class Link extends React.PureComponent<LinkProps, LinkState> {
     this.props.onMouseOut(this.props.linkData.source, this.props.linkData.target, evt);
   };
 
+  getLinkColor(): string[] | string | undefined| null  {
+    const { linkData, pathColorFunc } = this.props;
+  
+    if (typeof pathColorFunc === 'function') {
+       return  pathColorFunc(linkData);
+    }
+
+    return undefined;
+  }
+
+
+  getStrokeValue(linkColor, gradientId): string {
+    if (Array.isArray(linkColor)) {
+      return `url(#${gradientId})`;
+    } else {
+      return linkColor;
+    }
+  }
+
   render() {
-    const { linkData } = this.props;
+    const { linkData, pathColorFunc } = this.props;
+    const gradientId = getId();
+    const linkColor = this.getLinkColor();
+    const [gradientStartColor, gradientEndColor] = Array.isArray(linkColor) ? linkColor : [undefined, undefined];
+
     return (
-      <path
-        ref={l => {
-          this.linkRef = l;
-        }}
-        style={{ ...this.state.initialStyle }}
-        className={this.getClassNames()}
-        d={this.drawPath()}
-        onClick={this.handleOnClick}
-        onMouseOver={this.handleOnMouseOver}
-        onMouseOut={this.handleOnMouseOut}
-        data-source-id={linkData.source.id}
-        data-target-id={linkData.target.id}
-      />
+      <>
+      {gradientStartColor && <defs>
+        <linearGradient x1={linkData.source.x} y1={linkData.source.y} x2={linkData.target.x} y2={linkData.target.y} id={gradientId} gradientUnits="userStepOnUse">
```

Comment:
That's an invalid value for `gradientUnits`, you mean `userSpaceOnUse`:
```suggestion
        <linearGradient x1={linkData.source.x} y1={linkData.source.y} x2={linkData.target.x} y2={linkData.target.y} id={gradientId} gradientUnits="userSpaceOnUse">
```